src/Aspire.Hosting/api/Aspire.Hosting.cs

```diff
@@ -928,6 +1075,64 @@ public BeforeStartEvent(System.IServiceProvider services, DistributedApplication
         public System.IServiceProvider Services { get { throw null; } }
     }
 
+    public sealed partial class CertificateAuthorityCollectionAnnotation : IResourceAnnotation
+    {
+        public System.Collections.Generic.List<CertificateAuthorityCollection> CertificateAuthorityCollections { get { throw null; } }
+
+        public CertificateTrustScope? Scope { get { throw null; } }
+
+        public bool? TrustDeveloperCertificates { get { throw null; } }
+    }
+
+    public static partial class CertificateAuthorityCollectionResourceExtensions
+    {
+        public static IResourceBuilder<CertificateAuthorityCollection> AddCertificateAuthorityCollection(this IDistributedApplicationBuilder builder, string name) { throw null; }
+
+        public static IResourceBuilder<CertificateAuthorityCollection> WithCertificate(this IResourceBuilder<CertificateAuthorityCollection> builder, System.Security.Cryptography.X509Certificates.X509Certificate2 certificate) { throw null; }
+
+        public static IResourceBuilder<CertificateAuthorityCollection> WithCertificates(this IResourceBuilder<CertificateAuthorityCollection> builder, System.Collections.Generic.IEnumerable<System.Security.Cryptography.X509Certificates.X509Certificate2> certificates) { throw null; }
+
+        public static IResourceBuilder<CertificateAuthorityCollection> WithCertificates(this IResourceBuilder<CertificateAuthorityCollection> builder, System.Security.Cryptography.X509Certificates.X509Certificate2Collection certificates) { throw null; }
+
+        public static IResourceBuilder<CertificateAuthorityCollection> WithCertificatesFromFile(this IResourceBuilder<CertificateAuthorityCollection> builder, string pemFilePath, System.Func<System.Security.Cryptography.X509Certificates.X509Certificate2, bool>? filter = null) { throw null; }
+
+        public static IResourceBuilder<CertificateAuthorityCollection> WithCertificatesFromStore(this IResourceBuilder<CertificateAuthorityCollection> builder, System.Security.Cryptography.X509Certificates.StoreName storeName, System.Security.Cryptography.X509Certificates.StoreLocation storeLocation, System.Func<System.Security.Cryptography.X509Certificates.X509Certificate2, bool>? filter = null) { throw null; }
+    }
+
+    public sealed partial class CertificateTrustConfigurationCallbackAnnotation : IResourceAnnotation
+    {
+        public CertificateTrustConfigurationCallbackAnnotation(System.Func<CertificateTrustConfigurationCallbackAnnotationContext, System.Threading.Tasks.Task> callback) { }
+
+        public System.Func<CertificateTrustConfigurationCallbackAnnotationContext, System.Threading.Tasks.Task> Callback { get { throw null; } }
+    }
+
+    public sealed partial class CertificateTrustConfigurationCallbackAnnotationContext
```

Comment:
@danegsta - Does this need an IServiceProvider on it?

Reply:
It receives the ExecutionContext, which has a ServiceProvider.